datahub-web-react/src/app/permissions/policy/policyUtils.ts

```diff
@@ -126,3 +127,106 @@ export const setFieldValues = (
     }
     return { ...filter, criteria: [...restCriteria, createCriterion(resourceFieldType, fieldValues)] };
 };
+
+
+/**
+ * Add an entry to the ListPolicies cache.
+ */
+export const addToListPoliciesCache = (client, newPolicy, pageSize, query) => {
+    // Read the data from our cache for this query.
+    const currData: ListPoliciesQuery = client.readQuery({
+        query: ListPoliciesDocument,
+        variables: {
+            input: {
+                start: 0,
+                count: pageSize,
+                query,
+            },
+        },
+    });
+
+    const completeNewPolicy = {
+        ...newPolicy,
+        actors:{
+            ...newPolicy?.actors,
+            roles : newPolicy?.actors?.roles || [],
+            resourceOwnersTypes: newPolicy?.actors?.resourceOwnersTypes || [],
+            resolvedOwnershipTypes: newPolicy?.actors?.resolvedOwnershipTypes || [],
+            resolvedUsers: newPolicy?.actors?.resolvedUsers || [],
+            resolvedGroups: newPolicy?.actors?.resolvedGroups || [],
+            resolvedRoles: newPolicy?.actors?.resolvedRoles || [],
+        },
+        resources : {
+            ...newPolicy.resources,
+            type: newPolicy?.resources?.type || '', // replace '' with the default value for type
+            allResources: newPolicy?.resources?.allResources || '',
+            resources: newPolicy?.resources?.resources || ''
+        }
+    };
+
+    // Add our new source into the existing list.
+    const newPolicies = [completeNewPolicy, ...(currData?.listPolicies?.policies || [])];
+    
+    // Write our data back to the cache.
+    client.writeQuery({
+        query: ListPoliciesDocument,
+        variables: {
+            input: {
+                start: 0,
+                count: pageSize,
+                query,
+            },
+        },
+        data: {
+            listPolicies: {
+                start: 0,
+                count: (currData?.listPolicies?.count || 0) + 1,
+                total: (currData?.listPolicies?.total || 0) + 1,
+                policies: newPolicies,
+            },
+        },
+    });
+};
+
+
+/**
+ * Remove an entry from the ListIngestionSources cache.
+ */
+export const removeFromListPoliciesCache = (client, urn, page, pageSize, query) => {
```

Comment:
we'll need another method inside of here for updatePolicyInCache

Reply:
done